src/Hl7.Fhir.Base/Specification/Terminology/MultiTerminologyService.cs

```diff
@@ -78,6 +163,22 @@ public async Task<Parameters> ValueSetValidateCode(Parameters parameters, string
 
             throw new InvalidOperationException("We should never have come here");
         }
+
+        private IEnumerable<ITerminologyService> preferredService(Parameters parameters)
+        {
+            var valueSetUrl = new ValidateCodeParameters(parameters).Url?.Value;
```

Comment:
This function validates the parameters and then gets the right ITerminologyService - this sounds like a function with "low cohesion", since these are two completely different tasks.  E.g. what if the calling function also needs to get other parameters from the Parameters?  It would need to validate the parameters before it uses them, right?  So, I think you should separate the concerns of validating the parameters and getting the right iterminologyservi ce.